chore(http): Delete "Digest" and "Want-Digest", improve digest pages (#36879)
mdn/content: files/en-us/web/http/headers/repr-digest/index.md

{{HTTPSidebar}}

The HTTP **`Repr-Digest`** {{Glossary("Request header", "request")}} and {{Glossary("Response header", "response header")}} provides a {{Glossary("digest")}} of the selected representation of the target resource.

> A `Repr-Digest` is most suited to validate the integrity of partial or multipart messages against the full selected representation.

Reviewer: So, I wanted to change it like this:

```suggestion
A `Repr-Digest` can be used to validate the integrity of partial or multipart messages against the full selected representation.
```

Now I'd like you to tell me what that means ? :-). How can you validate a partial message using this information. The answer is probably you can't. My take is that this is only really useful once you have the whole message, and then it tells you that all the parts were OK.

Author: Yeah exactly, and I think the language we had was unnecessarily complex in some places,  what's becoming more clear to me from re-reading is that in Repr-Digest, the parts that you're receiving are not verified, but you know the whole resource is.

If you get a resource in two parts, you're not interested in the integrity of each arbitrary part as long as the whole thing is from something you can verify. IMO this is mostly interesting for range requests.

> The HTTP **`Repr-Digest`** {{Glossary("Request header", "request")}} and {{Glossary("Response header", "response header")}} provides a {{Glossary("digest")}} of the selected representation of the target resource.
> A `Repr-Digest` can be used to validate the integrity of partial or multipart messages against the whole selected representation.
> For example, in [range requests](/en-US/docs/Web/HTTP/Range_requests), a `Repr-Digest` in responses will always have the same value if only the requested byte ranges differ, whereas the {{HTTPHeader("Content-Digest")}} will be different for each range or part.

Reviewer: OK, taking another look at this. There is no reason to include this in a multipart message, and as you say, our language here gives the false impression it is somehow useful in this context.

```suggestion
The HTTP **`Repr-Digest`** {{Glossary("Request header", "request")}} and {{Glossary("Response header", "response header")}} provides a {{Glossary("digest")}} of the selected representation of the target resource.
It can be used validate the integrity of the whole selected representation once it has been received and reconstructed.
```

I removed this example. IT could be moved after/to belong with the text "The representation digest applies to the whole representation rather than the "

> For example, in [range requests](/en-US/docs/Web/HTTP/Range_requests), a `Repr-Digest` in responses will always have the same value if only the requested byte ranges differ, whereas the {{HTTPHeader("Content-Digest")}} will be different for each range or part.

Author: I think removing the sentence is fine seeing as we have examples below that cover different scenarios
